pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -4278,16 +4298,33 @@ func mapNodeNameToPrivateDNSName(nodeName types.NodeName) string {
 }
 
 // mapInstanceToNodeName maps a EC2 instance to a k8s NodeName, by extracting the PrivateDNSName
-func mapInstanceToNodeName(i *ec2.Instance) types.NodeName {
-	return types.NodeName(aws.StringValue(i.PrivateDnsName))
+func mapInstanceToNodeName(i *ec2.Instance, explicitNames bool) (types.NodeName, error) {
+	if explicitNames {
```

Comment:
I like the tags idea, but rather than differentiating between the old way (private DNS name) and the new way (tagged name) via a config option, I'd prefer if we can allow the new behavior in a backwards compatible way, without the need for a new config option.  I.e., default to private DNS name but use a tagged name if present on the node.